### PR TITLE
ddl: fix gofail data race

### PR DESCRIPTION
**Pull request by winkyao**
Fix #6897 data race

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: ddl/index.go [hunk 1]
```diff
@@ -690,7 +690,7 @@ func (w *addIndexWorker) run(d *ddlCtx) {
 
 		log.Debug("[ddl-reorg] got backfill index task:#v", task)
 		// gofail: var mockAddIndexErr bool
-		//if mockAddIndexErr && !gofailMockAddindexErrOnceGuard && w.id == 0 {
+		//if w.id == 0 && mockAddIndexErr && !gofailMockAddindexErrOnceGuard {
```
[the remark this answers is not in the document]
Comment (coocood) [reply ddl/index.go]: If the worker id is not 0, it will never access the `gofailMockAddindexErrOnceGuard` variable.

**Comment by shenli** [comment]
LGTM

**Pull request merged**